### PR TITLE
Initial saner tests

### PR DESCRIPTION
**Pull request by czh-orz**
This adds the initial tests for saner resources. It really only does the read and searches and will validate the resources returned from the searches. 

**Submitter:**
- [ ] This pull request describes why these changes were made
- [ ] Internal ticket for this PR:
- [ ] Internal ticket links to this PR
- [ ] Internal ticket is properly labeled (Community/Program)
- [ ] Internal ticket has a justification for its Community/Program label
- [ ] Code diff has been reviewed for extraneous/missing code
- [ ] Tests are included and test edge cases
- [ ] Tests/code quality metrics have been run locally and pass

**Reviewer 1:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

**Reviewer 2:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

File: generator/capability_statement_parser.rb [hunk 1]
```diff
@@ -120,93 +96,93 @@ def add_metadata_from_resources(metadata, resources)
...
+          return if sequence[:resource] == 'Observation'
+  
+          sequence[:required_concepts] = required_concepts.map do |concept|
+            concept['path']
+              .gsub("#{sequence[:resource]}.", '')
+              .gsub('[x]', 'CodeableConcept')
+          end
+        end
+  
+        def add_terminology_bindings(profile_definition, sequence)
```
Comment (radamson) [inline generator/capability_statement_parser.rb:179]: Ok to have, but do we still need this?

File: generator/generator_base.rb [hunk 2]
```diff
@@ -29,10 +29,22 @@ def load_resources
...
             new_resource = FHIR.from_contents(File.read(resource))
-            resource_by_path[resource_path(new_resource)] = new_resource_json
-            type = new_resource.class.name.demodulize
-            type = 'CapabilityStatement' if type == 'Conformance'
-            resources_by_type[type].push(new_resource_json)
+            if new_resource.id == 'searchParams'
+              new_resource.entry.each_with_index do |entry, index|
+                type = 'SearchParameter'
+                search_parameter_resource = entry.resource
+                next if resource_path(search_parameter_resource) == 'SearchParameter/SearchParameter-code'
```
Comment (radamson) [inline generator/generator_base.rb:36]: we'll want to make sure we can handle this eventually

File: generator/saner/interaction_test.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    module InteractionTest
+      def create_interaction_test(sequence, interaction)
+        return if interaction[:code] == 'read'
+
+        test_key = :"#{interaction[:code]}_interaction"
+        interaction_test = {
+          tests_that: "Server returns correct #{sequence[:resource]} resource from #{sequence[:resource]} #{interaction[:code]} interaction",
+          key: test_key,
+          index: sequence[:tests].length + 1,
+          link: 'https://www.hl7.org/fhir/us/core/CapabilityStatement-us-core-server.html',
```
Comment (radamson) [inline generator/saner/interaction_test.rb:13]: still some us core artifacts we should get rid of.

File: lib/app/sequence_base.rb [hunk 4]
```diff
@@ -529,8 +529,9 @@ def validate_reply_entries(resources, search_params)
         resources.each do |resource|
           # This checks to see if the base resource conforms to the specification
           # It does not validate any profiles.
-          resource_validation_errors = Inferno::RESOURCE_VALIDATOR.validate(resource, versioned_resource_class)
-          assert resource_validation_errors[:errors].empty?, "Invalid #{resource.resourceType}: #{resource_validation_errors[:errors].join("\n* ")}"
+
+          # resource_validation_errors = Inferno::RESOURCE_VALIDATOR.validate(resource, versioned_resource_class)
```
Comment (radamson) [inline lib/app/sequence_base.rb:533]: What was the reason for commenting this out?

File: lib/app/utils/validation.rb [hunk 5]
```diff
@@ -29,6 +29,8 @@ def self.definitions
       json = File.read(definition)
       version = VERSION_MAP[JSON.parse(json)['fhirVersion']]
       resource = get_resource(json, version)
+      next if resource.id == 'searchParams'
```
Comment (radamson) [inline lib/app/utils/validation.rb:32]: was it breaking without this if?
